internal/pkg/skuba/kubernetes/nodes.go

```diff
@@ -61,18 +59,47 @@ func IsControlPlane(node *v1.Node) bool {
 	return isControlPlane
 }
 
-func DrainNode(node *v1.Node, drainTimeout time.Duration) error {
-	// Drain node (shelling out, FIXME after https://github.com/kubernetes/kubernetes/pull/72827 can be used [1.14])
-	cmd := exec.Command("kubectl",
-		fmt.Sprintf("--kubeconfig=%s", skuba.KubeConfigAdminFile()),
-		"drain", "--delete-local-data=true", "--force=true", "--ignore-daemonsets=true", fmt.Sprintf("--timeout=%s", drainTimeout.String()), node.ObjectMeta.Name)
+func DrainNode(client kubernetes.Interface, node *v1.Node, drainTimeout time.Duration) error {
+	policyGroupVersion, err := kubectldrain.CheckEvictionSupport(client)
+	if err != nil {
+		return errors.Wrap(err, "could not get policy group version")
+	}
+
+	newCordon := kubectldrain.NewCordonHelper(node)
+	newCordon.UpdateIfRequired(true)
+	err, patchErr := newCordon.PatchOrReplace(client)
+	if err != nil {
+		return errors.Wrap(err, "failed to update node status")
+	}
+	if patchErr != nil {
+		return errors.Wrap(patchErr, "failed to patch node status")
+	}
 
-	if err := cmd.Run(); err != nil {
-		klog.V(1).Infof("could not drain node %s, aborting (use --force if you want to ignore this error)", node.ObjectMeta.Name)
-		return err
-	} else {
-		klog.V(1).Infof("node %s correctly drained", node.ObjectMeta.Name)
+	drainer := &kubectldrain.Helper{
+		Client:              client,
+		Force:               true,
+		IgnoreAllDaemonSets: true,
+		DeleteLocalData:     true,
+		Timeout:             drainTimeout,
 	}
+	del, errs := drainer.GetPodsForDeletion(node.ObjectMeta.Name)
+	if errs != nil {
+		return errors.New(fmt.Sprintf("could not get pods for deletion %s", errs))
+	}
+	for _, pod := range del.Pods() {
+		if len(policyGroupVersion) > 0 {
```

Comment:
All pods in the same node should have the same eviction capability?

If yes, check `len(policyGroupVersion) > 0` before the loop `for _, pod := range del.Pods()` would have better performance.

Reply:
I prefer to keep it within the loop. There is a fail prove to delete pod after eviction in case eviction doesn't work.

Reply:
On the second look, I think you are right. I will change it. Thanks.